### PR TITLE
ledstrip: blue warning on compass failure

### LINKED ISSUE #650: Clear indication of compass failure
A number of compass-related issues appear recently. In case it happens in the field, there is no clear indication to the pilot that some navigational features (including emergency RTH) may not work.

For GPS and BARO we may observe telemetry and OSD reports. But there's no such thing for MAG.

What would be the best indication method:
- re-using some telemetry fields (e.g. GPS related)?
- led-strip error indication?
- audible alert (buzzer)?
- arm-block (sufficient only for boot-time failure)?
- all the above?

### PR DESCRIPTION
**Pull request by krzysztofmatula**
- added new generic ledstrip warning: HW_ERROR (BLUE-BLACK)
- HW_ERROR warning set on compass failure (on init or in runtime)

fixes #650.

### CODE REVIEW

File: src/main/io/ledstrip.c [hunk 1]
```diff
@@ -535,6 +537,10 @@ static void applyLedWarningLayer(bool updateNow, uint32_t *timer)
                 warningFlags |= 1 << WARNING_FAILSAFE;
             if (!ARMING_FLAG(ARMED) && !ARMING_FLAG(OK_TO_ARM))
                 warningFlags |= 1 << WARNING_ARMING_DISABLED;
+#ifdef MAG
+            if (masterConfig.mag_hardware != MAG_NONE && !compassIsWorking())
```
Comment (digitalentity) [inline src/main/io/ledstrip.c]: I would make it a generic function i.e. `isHardwareFailing` so we can actually see if any of the hardware is not functioning properly (i.e. GPS is configured but not detected etc)

File: src/main/sensors/compass.c [hunk 2]
```diff
@@ -70,6 +70,11 @@ bool compassInit(int16_t magDeclinationFromConfig)
     return ret;
 }
 
+bool compassIsWorking(void)
```
Comment (digitalentity) [inline src/main/sensors/compass.c]: The check should probably be smarter than this. Please make it a generic function, we'll extend it further.

**Review by digitalentity — COMMENTED** [review]
Generally, I like the idea. If sensor is not set to "NONE" it must be detected and functional. All non-present sensors should be set to "NONE".

We should also report non-present and not-detected sensors via MSP and light up not detected or failing sensors with red in Configurator.

**Comment by digitalentity** [comment]
I've changed a base of the branch so we can collaborate and make this a generic hardware failure indication system and then merge.

File: src/main/sensors/compass.c [hunk 3]
```diff
@@ -70,6 +70,11 @@ bool compassInit(int16_t magDeclinationFromConfig)
     return ret;
 }
 
+bool isCompassHealthy(void)
+{
+    return (magADC[X] != 0) && (magADC[Y] != 0) && (magADC[Z] != 0);
```
Comment (krzysztofmatula) [inline src/main/sensors/compass.c:75]: Not sure if all-nonzero or rather any-nonzero should indicate healthy status.
The code above is copied from imu.c

Comment (digitalentity) [reply src/main/sensors/compass.c:75]: Great, thanks! I'll take it from here.

**Pull request merged**